<a href="https://colab.research.google.com/github/Narendernayak/Google_data_analysis/blob/master/data%20analysis%20(assenture).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd
import seaborn as ns


/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


#### Load dataset

In [0]:
data1 = pd.read_csv('M3+Screen+Traffic+Data+vFinal.csv')                                    
data2 = pd.read_csv('M3+Individual+Login+Data+vFinal.csv')

In [0]:
data1.head()

Unnamed: 0 Unnamed: 1                         Unnamed: 2  \
0  Overall Site        NaN  Impressions (count, overall site)   
1           NaN        NaN         Bounce rate (landing page)   
2           NaN        NaN                    Unique visitors   
3           NaN        NaN            Average pages per visit   
4           NaN        NaN             Successful Submissions   

                                         Definitions 1-May-19 2-May-19  \
0  Individual visits to overall site, including r...      133      147   
1  Number of students who access landing page and...      53%      48%   
2                   Number of visits from unique IPs      125      129   
3        Average number of pages accessed per log-in     2.00     1.67   
4      Percentage of submissions that are successful      47%      52%   

  3-May-19 4-May-19 5-May-19 6-May-19  ... 6-Jun-19 7-Jun-19 8-Jun-19  \
0      122      112      130      138  ...      129      136      150   
1      55%      54%      53%      54%  ...      51%      51%      17%   
2      118      106      121      122  ...      119      128      105   
3     1.33     2.00     1.33     1.00  ...     1.33     1.33     3.33   
4      47%      53%      42%      57%  ...      50%      46%      80%   

  9-Jun-19 10-Jun-19 11-Jun-19 12-Jun-19 13-Jun-19 14-Jun-19 Unnamed: 49  
0      142       187       166       139       154       161         NaN  
1      18%       16%       16%       17%       18%       16%         NaN  
2      103       104       101       101       102       102         NaN  
3     4.00      3.33      3.00      3.67      4.00      3.67         NaN  
4      98%       98%       93%       95%       86%       93%         NaN  

[5 rows x 50 columns]

In [0]:
data2.head()

Log-In date  Student login ID        University Logged-in duration  \
0    5/1/2019               6563              TSU           01:38:17   
1    5/1/2019              23287               RU           00:50:44   
2    5/1/2019           82074664              ILU           01:03:12   
3    5/1/2019             426932  Le Dean College           00:36:27   
4    5/1/2019              42189               RU           00:47:21   

  Timed out due to dormancy  Documents Uploaded (Number)  \
0                        No                            0   
1                       Yes                            0   
2                       Yes                            4   
3                       Yes                            1   
4                        No                            2   

   Documents Uploaded (file size, Mb) Submission  
0                                0.00         No  
1                                0.00         No  
2                               13.82        Yes  
3                               12.94        Yes  
4                                9.26        Yes

In [0]:
data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 50 columns):
Unnamed: 0     3 non-null object
Unnamed: 1     5 non-null object
Unnamed: 2     30 non-null object
Definitions    30 non-null object
1-May-19       29 non-null object
2-May-19       29 non-null object
3-May-19       29 non-null object
4-May-19       29 non-null object
5-May-19       29 non-null object
6-May-19       29 non-null object
7-May-19       29 non-null object
8-May-19       29 non-null object
9-May-19       29 non-null object
10-May-19      29 non-null object
11-May-19      29 non-null object
12-May-19      29 non-null object
13-May-19      29 non-null object
14-May-19      29 non-null object
15-May-19      29 non-null object
16-May-19      29 non-null object
17-May-19      29 non-null object
18-May-19      29 non-null object
19-May-19      29 non-null object
20-May-19      29 non-null object
21-May-19      29 non-null object
22-May-19      29 non-null object
23-May-19      

In [0]:
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5847 entries, 0 to 5846
Data columns (total 8 columns):
Log-In date                           5847 non-null object
Student login ID                      5847 non-null int64
University                            5847 non-null object
Logged-in duration                    5847 non-null object
Timed out due to dormancy             5847 non-null object
Documents Uploaded (Number)           5847 non-null int64
Documents Uploaded (file size, Mb)    5847 non-null float64
Submission                            5847 non-null object
dtypes: float64(1), int64(2), object(5)
memory usage: 365.6+ KB


#### Q.1 Since the most recent changes went live at the end of June 7th, how many students have accessed the site?

In [0]:
cols = data1.columns
print(cols)

Index(['Unnamed: 0', 'Unnamed: 1', 'Unnamed: 2', 'Definitions', '1-May-19',
       '2-May-19', '3-May-19', '4-May-19', '5-May-19', '6-May-19', '7-May-19',
       '8-May-19', '9-May-19', '10-May-19', '11-May-19', '12-May-19',
       '13-May-19', '14-May-19', '15-May-19', '16-May-19', '17-May-19',
       '18-May-19', '19-May-19', '20-May-19', '21-May-19', '22-May-19',
       '23-May-19', '24-May-19', '25-May-19', '26-May-19', '27-May-19',
       '28-May-19', '29-May-19', '30-May-19', '31-May-19', '1-Jun-19',
       '2-Jun-19', '3-Jun-19', '4-Jun-19', '5-Jun-19', '6-Jun-19', '7-Jun-19',
       '8-Jun-19', '9-Jun-19', '10-Jun-19', '11-Jun-19', '12-Jun-19',
       '13-Jun-19', '14-Jun-19', 'Unnamed: 49'],
      dtype='object')


In [0]:
# our target is to find how many students have accessed the site between 8 june 19 and 14 june 19
ls = data1.iloc[0].values  #first row of dataset (number of individual accessed the site)
num_students = np.sum(ls[42:49].astype(int)) #between 8-jun-19 and 14-jun-19

In [0]:
print(num_students)

1099


So, 1099 students have visited the site since modification in site, which is at the end of 7-jun-19

#### Q.2 I had an email from the Student Support Team at UAN – apparently their students are having problems with the site? Can you see if there’s a particular problem there please?

In [0]:
data2.head()

Log-In date  Student login ID        University Logged-in duration  \
0    5/1/2019               6563              TSU           01:38:17   
1    5/1/2019              23287               RU           00:50:44   
2    5/1/2019           82074664              ILU           01:03:12   
3    5/1/2019             426932  Le Dean College           00:36:27   
4    5/1/2019              42189               RU           00:47:21   

  Timed out due to dormancy  Documents Uploaded (Number)  \
0                        No                            0   
1                       Yes                            0   
2                       Yes                            4   
3                       Yes                            1   
4                        No                            2   

   Documents Uploaded (file size, Mb) Submission  
0                                0.00         No  
1                                0.00         No  
2                               13.82        Yes  
3                               12.94        Yes  
4                                9.26        Yes

In [0]:
uan_data = data2[data2['University'] == 'UAN']

In [0]:
uan_data

Log-In date  Student login ID  University Logged-in duration  \
220     5/2/2019            1672705        UAN           01:59:31   
3509   5/28/2019            1917311        UAN           01:17:50   
4267    6/3/2019            1002005        UAN           00:53:18   
5436   6/11/2019            5304074        UAN           00:38:56   
5558   6/12/2019            2727225        UAN           00:21:19   
5722   6/14/2019            9742822        UAN           01:57:29   
5820   6/14/2019            8052184        UAN           01:38:48   

     Timed out due to dormancy  Documents Uploaded (Number)  \
220                         No                            0   
3509                        No                            0   
4267                        No                            0   
5436                        No                            9   
5558                        No                            7   
5722                        No                            7   
5820                        No                           15   

      Documents Uploaded (file size, Mb) Submission  
220                                 0.00         No  
3509                                0.00         No  
4267                                0.00         No  
5436                                9.74        Yes  
5558                               12.35        Yes  
5722                                6.84        Yes  
5820                               14.36        Yes

As you can see, there is no time out due ti dormancy and there is also not a problem of document submission. So, from this data, we can't find problems faced by UAN students.

#### Q.3 The previous claim journey was time-consuming, and a lot of students were abandoning it partway through. Are we seeing fewer 'abandoned' claims, and has the average interaction time for a successful, completed claim journey now dropped?

In [0]:
data1['Definitions'].values

array(['Individual visits to overall site, including repeats from the same IP',
       'Number of students who access landing page and do not progress further',
       'Number of visits from unique IPs',
       'Average number of pages accessed per log-in',
       'Percentage of submissions that are successful',
       'Number of individual visits to log-in screen, including repeats from the same IP',
       'Average time spent on page per log-in',
       'Percentage of users who are logged out of portal due to time spent dormant on home screen',
       'Number of students who access log-in screen and do not progress further',
       'Number of individual visits to home screen, including repeats from the same IP',
       'Average time users spend on page for each submission attempt',
       'Percentage of users who are logged out of portal due to time spent dormant on home screen',
       'Number of visits to Homescreen from unique IPs',
       'Number of individual visits to Personal 